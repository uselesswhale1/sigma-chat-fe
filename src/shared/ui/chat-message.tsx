import { Flex, Avatar, Heading, Box } from "@chakra-ui/react";
import { Message } from "../../shared/models";
import { PALLETE } from "../constants/color-pallete";

interface ChatMessageProps {
  message: Message;
  isOwnMessage: boolean;
}

export const ChatMessage = ({
  message,
  isOwnMessage,
}: ChatMessageProps): JSX.Element => {
  // TODO fix date transformation
  const creationDate = new Date(message.createdAt)
    .toLocaleString()
    .split(", ")[1];

  const messageDirectionStyle = isOwnMessage ? "row-reverse" : "initial";
  const messageBgStyle = isOwnMessage ? PALLETE.a : PALLETE.b;

  return (
    <Flex
      flex="1"
      gap="4"
      flexWrap="wrap"
      alignItems="flex-start"
      m={2}
      flexDirection={messageDirectionStyle}
    >
      <Avatar name={message.creator.name} />

      <Box bg={messageBgStyle} p={2} borderRadius={10}>
        <Heading
          display="flex"
          flexDirection={messageDirectionStyle}
          size="sm"
          children={message.creator.name}
        />

        <Box w={360}>{message.content}</Box>

        <Heading
          as="h6"
          size="xs"
          display="flex"
          flexDirection="row-reverse"
          children={creationDate}
        />
      </Box>
    </Flex>
  );
};
