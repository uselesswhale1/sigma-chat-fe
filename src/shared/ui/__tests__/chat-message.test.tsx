import { render } from "@testing-library/react";
import { ChatMessage } from "../chat-message";

describe("ChatMessage should", () => {
  const props = {
    message: {
      content: "content",
      creator: {
        id: "123",
        firstName: "mock-user-first-name",
        email: "mock-user-email",
        name: "mock-user-name",
        createdAt: "2024-06-07T20:10:04.000Z",
        chats: [],
      },
      chatId: "chatId",
      createdAt: "2024-06-07T20:10:04.000Z",
    },
    isOwnMessage: true,
  };

  it("match snapshot", () => {
    const { container } = render(<ChatMessage {...props} />);

    expect(container).toMatchSnapshot();
  });

  it("match snapshot when not own message", () => {
    const { container } = render(
      <ChatMessage {...props} isOwnMessage={false} />
    );

    expect(container).toMatchSnapshot();
  });
});
