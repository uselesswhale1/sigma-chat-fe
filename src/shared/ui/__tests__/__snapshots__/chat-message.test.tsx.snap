// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage should match snapshot 1`] = `
<div>
  <div
    class="css-12h6285"
  >
    <span
      class="chakra-avatar css-1ruhx4w"
    >
      <div
        aria-label="mock-user-name"
        class="chakra-avatar__initials css-0"
        role="img"
      >
        m
      </div>
    </span>
    <div
      class="css-g6e4dm"
    >
      <h2
        class="chakra-heading css-tz0gqg"
      >
        mock-user-name
      </h2>
      <div
        class="css-3mdty7"
      >
        content
      </div>
      <h6
        class="chakra-heading css-tz0gqg"
      >
        11:10:04 PM
      </h6>
    </div>
  </div>
</div>
`;

exports[`ChatMessage should match snapshot when not own message 1`] = `
<div>
  <div
    class="css-fndydb"
  >
    <span
      class="chakra-avatar css-1ruhx4w"
    >
      <div
        aria-label="mock-user-name"
        class="chakra-avatar__initials css-0"
        role="img"
      >
        m
      </div>
    </span>
    <div
      class="css-ze3jos"
    >
      <h2
        class="chakra-heading css-zfn86r"
      >
        mock-user-name
      </h2>
      <div
        class="css-3mdty7"
      >
        content
      </div>
      <h6
        class="chakra-heading css-tz0gqg"
      >
        11:10:04 PM
      </h6>
    </div>
  </div>
</div>
`;
